packages/core/src/index.ts

```diff
@@ -19,6 +19,7 @@ export type {
   NormalizedConfig,
   // Plugin Types
   RsbuildPlugin,
+  RsbuildPlugins,
```

Comment:
why do we need to export this type in the entry module?

Reply:
I believe that all types used by the config object should be exported. This way, upper toolkits like modern.js can extend or assemble the config type from scratch.

Reply:
This file only exports **public and documented** public API, there is a comment:

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/a683d906-4b3c-4a9e-9809-4bfa5ace34f1)